components/eventing-controller/pkg/env/nats_config.go

```diff
@@ -50,6 +50,8 @@ type NatsConfig struct {
 	// - new: When first consuming messages, the consumer starts receiving messages that were created
 	//   after the consumer was created.
 	JSConsumerDeliverPolicy string `envconfig:"JS_CONSUMER_DELIVER_POLICY" default:"new"`
+
+	EnableNewCRDVersion bool `envconfig:"ENABLE_NEW_CRD_VERSION" default:"false"`
```

Comment:
do we really need this flag here also ? isn't it sufficient to have it in the `config.go` ?

Reply:
we add it here since we do not want to change the struct of the `SubscriptionManager` for nats which currently takes in the natsconfig as an argument.

Reply:
Makes sense, keep it like this.